### PR TITLE
 Turn trailing implicit warning into an error 

### PR DESCRIPTION
**Pull request by SimonBoulier**
This PR turn the following warning:

```
Arguments nil [_].

Toplevel input, characters 0-18:
> Arguments nil [_].
> ^^^^^^^^^^^^^^^^^^
Warning: Argument number 0 is a trailing implicit so must be maximal
[set-maximal-deprecated,deprecated]
```

into the following error:
```
Arguments nil [_].

Toplevel input, characters 0-18:
> Arguments nil [_].
> ^^^^^^^^^^^^^^^^^^
Error: Argument A is a trailing implicit, so it can't be declared non
maximal. Please use { } instead of [ ].
```

Thiw was marked deprecated since Coq 8.10. This is done now to avoid introducing a new way of raising this deprecated warning in PR #11235.

- [ ] Added / updated test-suite
- [x] Corresponding documentation was added / updated (including any warning and error messages added / removed / modified).
- [x] Entry added in the changelog (see https://github.com/coq/coq/tree/master/doc/changelog#unreleased-changelog for details).

### CODE REVIEW

File: doc/sphinx/language/gallina-extensions.rst [hunk 1]
```diff
@@ -1728,11 +1728,11 @@ Declaring Implicit Arguments
    To know which are the implicit arguments of an object, use the
    command :cmd:`Print Implicit` (see :ref:`displaying-implicit-args`).
 
-.. warn:: Argument number @num is a trailing implicit so must be maximal.
+.. exn:: Argument id is a trailing implicit, so it can't be declared non maximal. Please use %{ %} instead of [ ].
```
Comment (maximedenes) [inline doc/sphinx/language/gallina-extensions.rst]: Shouldn't `id` be `@ident` or something like that? @coq/doc-maintainers will know better.

File: interp/impargs.ml [hunk 2]
```diff
@@ -648,9 +648,8 @@ let maybe_declare_manual_implicits local ref ?enriching l =
 
 (* TODO: either turn these warnings on and document them, or handle these cases sensibly *)
```
Comment (maximedenes) [inline interp/impargs.ml]: Is the comment now outdated?

Comment (SimonBoulier) [reply interp/impargs.ml]: Indeed thanks.

File: doc/sphinx/language/gallina-extensions.rst (hunk 1 above)
Comment (SimonBoulier) [reply doc/sphinx/language/gallina-extensions.rst]: Seems to work indeed.

**Comment by SimonBoulier** [comment]
For me it is ready modulo CI.

File: doc/sphinx/language/gallina-extensions.rst (hunk 1 above)
Comment (Zimmi48) [reply doc/sphinx/language/gallina-extensions.rst]: Actually currently we do not parse such notations in warning / error messages. But feel free to still use them. The pattern is so common that we should absolutely parse those.

**Comment by Zimmi48** [comment]
Have the overlays already be submitted / merged in the corresponding projects?

File: doc/changelog/02-specification-language/11368-trailing_implicit_error.rst [hunk 3]
```diff
@@ -0,0 +1,6 @@
+- **Changed:**
+  The warning raised when a trailing implicit is declared to be non maximally
+  inserted (with the command `Arguments`) has been turned into an error.
```
Comment (Zimmi48) [inline doc/changelog/02-specification-language/11368-trailing_implicit_error.rst]: You could use the following:
```suggestion
  inserted (with the command cmd:`Arguments <Arguments (implicits)>`) has been turned into an error.
```

**Comment by SimonBoulier** [comment]
> Have the overlays already be submitted / merged in the corresponding projects?

No. I only created the branches. Should I create PRs?

**Comment by gares** [comment]
> No. I only created the branches. Should I create PRs?

yes

**Comment by SimonBoulier** [comment]
> No. I only created the branches. Should I create PRs?

Done!

File: doc/sphinx/language/gallina-extensions.rst (hunk 1 above)
Comment (Zimmi48) [reply doc/sphinx/language/gallina-extensions.rst]: Actually, I was wrong. We do parse these notations. It is just the warning / error index that does not display them correctly, which we should fix.

**Comment by SimonBoulier** [comment]
@herbelin Can this be merged?

**Review by herbelin — APPROVED** [review]
This is good to me.

I'm assigning.

Regarding the questions about extra implicit arguments beyond the last one due to coercions to Funclass, see #11327. 

If no further remarks, will merge tomorrow evening.

**Comment by ppedrot** [comment]
@herbelin you should probably merge now...

**Comment by herbelin** [comment]
Ah, yes, certainly... I forgot to assign, thanks for reminding.

**Comment by herbelin** [comment]
Actually, I maybe went too fast. The overlays being presumably backwards compatible, it would have been better to ask for merging the numerous overlays first. Sorry.

**Pull request merged**